### ISSUE TITLE
Missed check for GS support in Port Library for zLinux

### ISSUE THREAD

**Issue by dmitripivkine** [eclipse/openj9#1764]
We might miss check for availability of Guarded Storage support in Port Library for zLinux. This might be a reason for crash if OS does not support GS. Could we double check?

@ebadyano @fjeremic Is it a real issue? Should it be addressed before we release CS for zLinux?

**Comment by dmitripivkine** [comment]
@joransiu FYI

**Comment by fjeremic** [comment]
> Is it a real issue?

It does certainly seem to be an issue. I cannot locate the code which checks the `auxv` bits on Linux on z. I was certain we had this, but all I can find so far is the z/OS CVT bit checking.

**Comment by joransiu** [comment]
> It does certainly seem to be an issue. I cannot locate the code which checks the auxv bits on Linux on z. I was certain we had this, but all I can find so far is the z/OS CVT bit checking.

Yeah, I'm utterly confused here too.  Other features like Vector Facility would have OS dependencies as well, and I did not find queries for auxv and/or cpu features in `/proc/cpuinfo`.

**Comment by fjeremic** [comment]
It seems we do check `/proc/cpuinfo` for the vector support from here:

https://github.com/eclipse/openj9/blob/78560086e6911b7d0998aeb8118018a8afc22772/runtime/compiler/trj9/z/env/J9CPU.cpp#L401-L406

which calls:

https://github.com/eclipse/openj9/blob/78560086e6911b7d0998aeb8118018a8afc22772/runtime/compiler/trj9/z/env/J9CPU.cpp#L401-L406

We don't seem to have anything like this for Guarded Storage. In fact we probably shouldn't. We need to eliminate this kind of dodgy string checking code and just rely on the result of `auxv`, probably even before we check STFLE bits.

**Comment by SueChaplain** [comment]
Do we need to add anything to the 0.9.0 release notes for this please? What is the status w.r.t support on RHEL7.5 and Ubuntu 18.04.

**Comment by ebadyano** [comment]
https://www-03preprod.ibm.com/support/knowledgecenter/SSYKE2_8.0.0_test/com.ibm.java.vm.80.doc/docs/xgc.html RHEL 7.5 supports Guarded Storage, not sure about Ubuntu @joransiu Joran, do you know if it's supported on Ubuntu?

If -Xgc:concurrentScavenge is used on hardware that supports guarded storage with OS that doesn't, the symptom will be **stderr Fatal error: Failed to initialize thread; please disable Concurrent Scavenge.**

**Comment by joransiu** [comment]
@ebadyano : FYI.. the link above is a pre-production version of our upcoming KC update.   Ubuntu 18.04 does have the kernel support for Guarded Storage Facility.

**Comment by ebadyano** [comment]
> @ebadyano : Should we not default down to a warning and disable Concurrent Scavenge under the covers in such a scenario? I think that's what we currently do if you use -Xgc:concurrentScavenge on a z13 or older.
@joransiu We also give a warning and continue to run with Concurrent Scavenge disabled on zOS when we run on z14 but with an older zOS that doesn't support guarded storage. Due to missing OS check for Linux it won't do the right default but instead will give the error message and won't run, which I agree we should fix in the next update.

**Comment by joransiu** [comment]
Thanks @ebadyano .  So if I understand correctly, the issue is not a crash, but a Fatal Error.  After the fix, we'll have a warning, and default down to non-concurrent scavenge.  correct?

If that's the case, scenario (1) above is not a real concern... in some ways, it's arguably better feedback to the user that their stack is not up to spec.

**Comment by ebadyano** [comment]
@joransiu I tested the behavior and it looks like for zOS when we attempt to run with  -Xgc:concurrentScavenge on hardware or OS that doesn't support Guarded Storage we just default to run without concurrent scavenger, the only way to see that it didn't run is to check verbose gc. @amicic Aleks, could you confirm if that's what we want and do we want to add some kind of warning?
